analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -177,4 +175,23 @@ private ProgramState ProcessMethod(SymbolicContext context, IMethodSymbol method
         }
         return state;
     }
+
+    private static CollectionConstraint CollectionCreationConstraint(ProgramState state, IObjectCreationOperationWrapper objectCreation) =>
+        objectCreation.Arguments.SingleOrDefault(x => x.ToArgument().Parameter.Type.DerivesOrImplements(KnownType.System_Collections_IEnumerable)) is { } collectionArgument
+            ? state[collectionArgument]?.Constraint<CollectionConstraint>()
+            : CollectionConstraint.Empty;
```

Comment:
This one is not changed, I just moved nonstatic ProcessMethod above the static ones and couldn't get the commit history play nicely about that